File: modules/expo-video/ios/ExpoVideoModule.swift
import ExpoModulesCore

@available(iOS 14.0, *)
public class ExpoVideoModule: Module {
    private weak var mainView: ExpoVideoView?
    public func definition() -> ModuleDefinition {
        
        Name("ExpoVideo")
        
        View(ExpoVideoView.self) {
            Prop("url") { (view: ExpoVideoView, prop: String) in
                self.mainView = view
                view.viewModel.setUrl(prop)
            }
        }
    }
}

File: modules/expo-video/android/src/main/java/expo/modules/video/ExpoVideoModule.kt
package expo.modules.video

import VideoUrlRecord
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.video.models.VideoUrlData

class ExpoVideoModule : Module() {
  private var mainView: ExpoVideoView? = null
  override fun definition() = ModuleDefinition {
    Name("ExpoVideo")

    View(ExpoVideoView::class) {
      Prop("url") { view: ExpoVideoView, prop: String ->
        mainView = view;
        view.viewModel.updateUrl(prop)
      }
    }
  }
}
